src/VirtoCommerce.Platform.Security/HttpContextUserResolver.cs

```diff
@@ -15,7 +16,7 @@ public HttpContextUserResolver(IHttpContextAccessor httpContextAccessor)
 
         public string GetCurrentUserName()
         {
-            var result = Thread.CurrentPrincipal?.Identity?.Name ?? "unknown";
+            var result = Thread.GetData(Thread.GetNamedDataSlot(ThreadSlotNames.USER_NAME)) as string ?? "unknown";
```

Comment:
Probably we could add a constant for "unknown" that could be used everywhere (here and in ApplicationBuilderExtensions).